### PR TITLE
ENG-183: add starter pack check wallet flow

### PR DESCRIPTION
**Pull request by stormcloud266**
Adds starter pack minting functionality

### CODE REVIEW

File: pages/claim/[puzzleName].tsx [hunk 1]
```diff
@@ -45,33 +45,49 @@ const buttonData = [
...
   const checkIfClaimed = async (account: string) => {
-    const url = `/api/minter/achievement?account=${account}&tokenId=${tokenId.toString()}`;
+    const url = `/api/minter/check-claimed?account=${account}&tokenId=${tokenId.toString()}`;
 
     const response = await fetch(url);
     if (response.ok) return (await response.json()).claimed;
     else throw await response.text();
   };
 
+  const verify = async (account: string, tokenId: number, chain: number) => {
```
Comment (illepic) [inline pages/claim/[puzzleName].tsx:77]: We gotta tune this up soon.

File: pages/pack/[packName].tsx [hunk 2]
```diff
@@ -30,22 +42,120 @@ interface PageParams {
...
+  const nftId = pack.nftId;
+
   const width = useCurrentWidth();
   const layout = width < 640 ? PuzzleLayoutType.List : PuzzleLayoutType.Grid;
+  const [chain, setChain] = useState<number | undefined>();
+  const [claimed, setClaimed] = useState<boolean>(false);
+  const [account, setAccount] = useState<string>("");
+  const [message, setMessage] = useState<string>("");
+  const [signature, setSignature] = useState<string>("");
+  const [loading, setLoading] = useState<boolean>(false);
```
Comment (illepic) [inline pages/pack/[packName].tsx:70]: We absolutely have to move this to xstate next.

File: pages/pack/[packName].tsx [hunk 3]
```diff
@@ -30,22 +42,120 @@ interface PageParams {
...
+  //         "You do not have the required NFTS on this chain. Please ensure you have completed the above puzzles and are on the correct chain."
+  //       );
+  //     setLoading(false);
+  //     return data.claimed;
+  //   } else {
+  //     setLoading(false);
+  //     setMessage("Something went wrong. Please try again.");
+  //     throw await response.text();
+  //   }
+  // };
```
Comment (illepic) [inline pages/pack/[packName].tsx]: Nuke commented out code.

File: pages/pack/[packName].tsx [hunk 4]
```diff
@@ -80,21 +191,98 @@ const PacksPage: NextPage<PageProps> = ({ puzzles, puzzlesNftIds, pack }) => {
...
+                        rel="noopener noreferrer"
+                        href={`${
+                          chain === ETH_CHAIN_ID
+                            ? openseaLink
+                            : chain === AVAX_CHAIN_ID
+                            ? joePegsLink
+                            : chain === POLYGON_CHAIN_ID
+                            ? openseaPolygonLink
+                            : openseaLink
+                        }${nftId}`}
```
Comment (illepic) [inline pages/pack/[packName].tsx:199]: we gotta turn this into a helper function, but can later

**Review by illepic — CHANGES_REQUESTED** [review]

**Review by illepic — APPROVED** [review]

**Review by chair28980 — APPROVED** [review]
LGTM 🍏

**Pull request merged**